connect spring-cloud

1EVILGUN1 opened the pull request:
перенос модуля main-service в модуль core, добавление модуля interaction-api для dto,model и клиентов. Добавление модуля infra для хранения config,discovery,gateway модулей stats, core и их дочерние модулю. Также доработаны логирование во всех класса main-service, убраны магические числа и перенесены ввиде констант в начало класса

1EVILGUN1 committed:
connect spring-cloud

перенос модуля main-service в модуль core, добавление модуля interaction-api для dto,model и клиентов.
Добавление модуля infra для хранения config,discovery,gateway модулей stats, core и их дочерние модулю. Также доработаны логирование во всех класса main-service, убраны магические числа и перенесены ввиде констант в начало класса

VadimZharkov commented on infra/config-server/src/main/resources/config/stats/stats-server/application.yml:
```diff
@@ -0,0 +1,17 @@
+server:
+  port: 9090
```
Лучше 0, чтобы тоже запускался на случайном порту

VadimZharkov commented on stats/stats-server/src/main/resources/application.yml:
```diff
@@ -0,0 +1,27 @@
+    instance-id: "${spring.application.name}:9090"
+    leaseRenewalIntervalInSeconds: 10
+
+server:
```
Это лишнее, т.к. порт возьмется из сервиса конфигурации

VadimZharkov commented on core/main-service/src/main/resources/application.yml:
```diff
@@ -0,0 +1,27 @@
+    instance-id: "${spring.application.name}:${random.value}"
+    leaseRenewalIntervalInSeconds: 10
+
+server:
```
Это лишнее, т.к. порт возьмется из сервиса конфигурации

VadimZharkov reviewed (changes requested):
Хорошая работа, правильно создана структура проекта. Можно немного улучшить в части конфигурации портов

1EVILGUN1 committed:
correction of comments review

замена порта 9090 на 0 указывание порта в docker-compose добавление в StatsClient класса конфигуратора с loadbalancer

1EVILGUN1 committed:
update statsclient

перенос resttemplateconfig в отдельный класс
